workflows/metapep.nf

```diff
@@ -0,0 +1,328 @@
+        // the channel is branched into buffer or unbuffer depending on the file count
+        // therefore one of both channels will be empty
+        SPLIT_PRED_TASKS.out.ch_epitope_prediction_chunks.flatten().count()
+            .combine(PREDICT_EPITOPES.out.ch_epitope_predictions.toSortedList().flatten())
+            .merge(PREDICT_EPITOPES.out.ch_epitope_prediction_warnings.toSortedList().flatten())
+            .branch{count, predictions_file, warnings_file ->
+                buffer: count > params.pred_buffer_files
+                    return [predictions_file, warnings_file]
+                unbuffered: count <= params.pred_buffer_files
+                    return [predictions_file, warnings_file]
+            }.set { ch_pred_merge_input }
+
+        // remap the individual files to individual channels to make the buffering work in later step
+        // unbuffered needs to be remapped to individual channels to make the mixing of merge buffer and merge input work
+        ch_pred_merge_input.buffer.multiMap{predictions_file, warnings_file ->
+            predictions: predictions_file
+            warnings: warnings_file
+        }.set { ch_predictions_mergebuffer_input }
+
+        ch_pred_merge_input.unbuffered.multiMap{predictions_file, warnings_file ->
+            predictions: predictions_file
+            warnings: warnings_file
+        }.set { ch_predictions_unbuffered }
+
+        // Process is only used when files exceed the buffer files parameter (default 1000) -> May generates issues for slurm if larger
+        MERGE_PREDICTIONS_BUFFER (
+            ch_predictions_mergebuffer_input.predictions.buffer(size: params.pred_buffer_files, remainder: true),
+            ch_predictions_mergebuffer_input.warnings.buffer(size: params.pred_buffer_files, remainder: true)
+        )
+        ch_versions = ch_versions.mix(MERGE_PREDICTIONS_BUFFER.out.versions)
+
+        // Mix the output of the merge predictions buffer channel and merge predictions channel (one of them will be empty)
+        ch_merge_predictions_input_pred = MERGE_PREDICTIONS_BUFFER.out.ch_predictions_merged_buffer.mix(ch_predictions_unbuffered.predictions)
+        ch_merge_predictions_input_warn = MERGE_PREDICTIONS_BUFFER.out.ch_prediction_warnings_merged_buffer.mix(ch_predictions_unbuffered.warnings)
+
+        MERGE_PREDICTIONS (
+            ch_merge_predictions_input_pred.collect(sort: { it.baseName }),
+            ch_merge_predictions_input_warn.collect(sort: { it.baseName })
+        )
+        ch_versions = ch_versions.mix(MERGE_PREDICTIONS.out.versions)
+
+        //
+        // MODULE: Plot score distributions
+        //
+        PREPARE_SCORE_DISTRIBUTION (
+            MERGE_PREDICTIONS.out.ch_predictions,
+            GENERATE_PEPTIDES.out.ch_proteins_peptides,
+            GENERATE_PROTEIN_AND_ENTITY_IDS.out.ch_entities_proteins,
+            FINALIZE_MICROBIOME_ENTITIES.out.ch_microbiomes_entities,
+            PROCESS_INPUT.out.ch_conditions,
```

Comment:
These 3 PROCESS_INPUT informations will can be removed when you use a meta map in the future :)

Reply:
True I'll add that to an Issue as well!